types/underscore/index.d.ts

```diff
@@ -813,54 +815,40 @@ declare module _ {
         unique: UnderscoreStatic['uniq'];
 
         /**
-        * Merges together the values of each of the arrays with the values at the corresponding position.
-        * Useful when you have separate data sources that are coordinated through matching array indexes.
-        * If you're working with a matrix of nested arrays, zip.apply can transpose the matrix in a similar fashion.
-        * @param arrays The arrays to merge/zip.
-        * @return Zipped version of `arrays`.
-        **/
-        zip(...arrays: any[][]): any[][];
-
-        /**
-        * @see _.zip
-        **/
-        zip(...arrays: any[]): any[];
-
-        /**
-        * The opposite of zip. Given a number of arrays, returns a series of new arrays, the first
-        * of which contains all of the first elements in the input arrays, the second of which
-        * contains all of the second elements, and so on. Use with apply to pass in an array
-        * of arrays
-        * @param arrays The arrays to unzip.
-        * @return Unzipped version of `arrays`.
-        **/
-        unzip(...arrays: any[][]): any[][];
-
-        /**
-        * Converts arrays into objects. Pass either a single list of [key, value] pairs, or a
-        * list of keys, and a list of values.
-        * @param keys Key array.
-        * @param values Value array.
-        * @return An object containing the `keys` as properties and `values` as the property values.
-        **/
-        object<TResult extends {}>(
-            keys: _.List<string>,
-            values: _.List<any>): TResult;
+         * Merges together the values of each of the `lists` with the values at
+         * the corresponding position. Useful when you have separate data
+         * sources that are coordinated through matching list indexes. If
+         * you're working with a matrix of nested lists, this can be used to
+         * transpose the matrix.
```

Comment:
I took out the bit about `apply` because I felt it was a little outdated at this point; unless folks want to change the `this` context for a function, I think folks should ideally be using the array spread operator these days.

For some reason I thought that these were doing different stuff for some reason. Looking at the source for these, though, it looks like `zip` just builds an array and calls `unzip`, and there's no set of inputs to `zip` that I can't package into an array and get the same result from `unzip`. I guess that means in a super ideal world rather than assuming that `zip` is always being used to translate homogenous arrays into non-homogenous ones and vice versa for `unzip` I should ideally make it so that both of these can deal with both directions.

Reply:
Almost forgot to actually remove the requested part of these comments; I've done so now.